scripts/tools/zap/cipd_upload.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+import logging
+import os
+import shlex
+import subprocess
+import sys
+import tempfile
+
+import click
+
+sys.path.append(os.path.abspath(os.path.dirname(__file__)))
+from zap_download import DownloadReleasedZap  # noqa: E402 isort:skip
+
+
+try:
+    import coloredlogs
+    _has_coloredlogs = True
+except ImportError:
+    _has_coloredlogs = False
+
+# Supported log levels, mapping string values required for argument
+# parsing into logging constants
+__LOG_LEVELS__ = {
+    'debug': logging.DEBUG,
+    'info': logging.INFO,
+    'warn': logging.WARN,
+    'fatal': logging.FATAL,
+}
+
+
+def _ZapArchitectures():
+    """Iterates through required releases.
+
+    Returns a tuple:
+      - zap zip 'platform'
+      - zap zip 'arch'
+      - CIPD package tuple
+    """
+
+    yield 'linux', 'arm64', 'linux-arm64'
+    yield 'linux', 'x64', 'linux-amd64'
+    yield 'mac', 'arm64', 'mac-arm64'
+    yield 'mac', 'x64', 'mac-amd64'
+    yield 'win', 'x64', 'windows-amd64'
```

Comment:
This should just be a top-level list of tuples:

```
ZAP_ARCHITECTURES = [
   (a, b, c),
   (a, b, c),
]
```

Then instead of iterating over a generator, you iterate over a list.